docs/spark/how-to-guides/ubuntu-instructions.md

```diff
@@ -0,0 +1,249 @@
+---
+title: Build a .NET for Apache Spark application on Ubuntu
+description: Learn how to build your .NET for Apache Spark application on Ubuntu
+ms.date: 01/02/2020
+ms.topic: conceptual
+ms.custom: mvc,how-to
+---
+
+
+# Learn how to build your .NET for Apache Spark application on Ubuntu
+
+## Prerequisites
+
+If you already have all the prerequisites, skip to the [build](#build) steps below.
+
+  1. Download and install **[.NET Core 2.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/2.1)** or the **[.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1)** - installing the SDK adds the `dotnet` toolchain to your path.
+  2. Install **[OpenJDK 8](https://openjdk.java.net/install/)**. 
+
+     - You can use the following command:
+
+       ```bash
+       sudo apt install openjdk-8-jdk
+       ```
+
+     - Verify you are able to run `java` from your command-line.       
+
+       Sample java -version output:
+       
+       ```bash
+       openjdk version "1.8.0_191"
+       OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-2ubuntu0.18.04.1-b12)
+       OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
+       ```
+
+     - If you already have multiple OpenJDK versions installed and want to select OpenJDK 8, use the following command:
+
+       ```bash
+       sudo update-alternatives --config java
+       ```
+
+  3. Install **[Apache Maven 3.6.0+](https://maven.apache.org/download.cgi)**.
+
+     - Run the following command:
+
+       ```bash
+       mkdir -p ~/bin/maven
+       cd ~/bin/maven
+       wget https://www-us.apache.org/dist/maven/maven-3/3.6.0/binaries/apache-maven-3.6.0-bin.tar.gz
+       tar -xvzf apache-maven-3.6.0-bin.tar.gz
+       ln -s apache-maven-3.6.0 current
+       export M2_HOME=~/bin/maven/current
+       export PATH=${M2_HOME}/bin:${PATH}
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+
+     - Verify you are able to run `mvn` from your command-line       
+
+       Sample mvn -version output:
+       
+       ```
+       Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
+       Maven home: ~/bin/apache-maven-3.6.0
+       Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
+       Default locale: en, platform encoding: UTF-8
+       OS name: "linux", version: "4.4.0-17763-microsoft", arch: "amd64", family: "unix"
+       ```
+
+  4. Install **[Apache Spark 2.3+](https://spark.apache.org/downloads.html)**.
```

Comment:
Would we like to also explain to users the other versions of Apache Spark they could use? 

We used this line in the getting started tutorial doc: "Download and install Apache Spark. You'll need to select from version 2.3.* or 2.4.0, 2.4.1, 2.4.3, or 2.4.4 (.NET for Apache Spark is not compatible with other versions of Apache Spark)."

Reply:
Yes, we can mention the supported versions here. Will make that change, thanks!